Enhance MainPage UI and integrate audio service for motion detection

AbdulrahmanAhmeed opened the pull request:
- Added a settings button to the camera stream section in MainPage.xaml for user configuration.
- Implemented audio service integration in MainPage.xaml.cs to play sounds for motion detection events.
- Refactored MauiProgram.cs to register the audio service based on the platform.
- Improved error handling and logging for audio playback functionality.

AbdulrahmanAhmeed committed:
Enhance MainPage UI and integrate audio service for motion detection

- Added a settings button to the camera stream section in MainPage.xaml for user configuration.
- Implemented audio service integration in MainPage.xaml.cs to play sounds for motion detection events.
- Refactored MauiProgram.cs to register the audio service based on the platform.
- Improved error handling and logging for audio playback functionality.

elmerbool commented on IPCameraViewer/Platforms/Windows/AudioService.cs:
```diff
@@ -0,0 +1,97 @@
+
+        public void PlaySound(string? filePath)
+        {
+            if (string.IsNullOrWhiteSpace(filePath))
```
Use one return statement.

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -222,9 +253,82 @@ private void OnMotion(CameraStreamViewModel streamViewModel)
 
+        private void PlayMotionSound()
+        {
+            try
```
Use single return statement.

elmerbool commented on IPCameraViewer/SettingsPage.xaml.cs:
```diff
@@ -0,0 +1,159 @@
+            if (string.IsNullOrEmpty(this.selectedFilePath) || !File.Exists(this.selectedFilePath))
+            {
+                this.DisplayAlert(SettingsPage.ErrorTitle, SettingsPage.NoValidSoundFileMessage, SettingsPage.OkButtonText);
+                return;
```
Use single return statement.

AbdulrahmanAhmeed committed:
Refactor audio playback logic for improved error handling and clarity

- Simplified audio playback checks in MainPage.xaml.cs to ensure audio service is available before attempting to play sounds.
- Enhanced error handling in SettingsPage.xaml.cs for sound file validation and audio service accessibility.
- Improved the PlaySound method in AudioService.cs to handle various exceptions and ensure proper file path validation.
